mlrun/artifacts/manager.py

```diff
@@ -121,6 +122,26 @@ def __init__(
         self.input_artifacts = {}
         self.artifacts = {}
 
+    @staticmethod
+    def validate_artifact_location(item, path, body):
```

Comment:
Would change the name a bit to something like `ensure_artifact_source_file_exists`. Also, please add some comments here explaining the logic and why you do the exceptions that you do (if we have body, then no file is needed, model artifact is a directory etc.).